### PR TITLE
Upgrading amqp_client and rabbit_common to v3.7.3 to support Elixir v1.6.1

### PR DESCRIPTION
**Pull request by carl-al**
Upgrading amqp_client and rabbit_common to v3.7.3 which support officially support Elixir v1.6
I did have compilation errors when performing `mix deps.compile` which are now resolved

### CODE REVIEW

**Comment by carl-al** [comment]
@ono @pma for your perusal

**Comment by ono** [comment]
As I commented on other thread, I will look into it later this week. Your description is slightly misleading though. The current stable version works with Elixir 1.6. It is more to update rabbit_client/common and Travis CI configuration.

**Comment by carl-al** [comment]
@ono would you like me to update the stable version also?

**Comment by ono** [comment]
@carl-al nope. don't worry about these versioning work.

File: mix.exs [hunk 1]
```diff
@@ -22,13 +22,13 @@ defmodule AMQP.Mixfile do
   end
 
   def application do
-    [applications: [:logger, :amqp_client]]
+    [applications: [:logger, :rabbit_common, :amqp_client]]
```
Comment (ono) [inline mix.exs]: any reason you needed this change?

Comment (carl-al) [reply mix.exs]: I added this in an attempt to get the failing test to work, which failed. Apologies. Let me remove it.

Comment (carl-al) [reply mix.exs]: done @ono

**Comment by ono** [comment]
Let me double check. rabbit_common 3.6.14 doesn't work with Elixir 1.6? It's Erlang library and hard to imagine it's affected by Elixir version. If you have seen the issue, could you share it please?

I'm not against the idea updating rabbit_common/client but want to understand so that we can note it [our compatibility document](https://github.com/pma/amqp/wiki/Versions-and-Compatibilities).

Thanks!

**Comment by ono** [comment]
Pushed equivalent changes to master.
- https://github.com/pma/amqp/commit/e322d23fe86abf817c1ef6fe7562d4455f1c7733
- https://github.com/pma/amqp/commit/0c0544a5d39846a21efdeacb29912d2633d47175

Thanks for the PR though 👍

**Pull request closed without merging**